dev/benchmarks/microbenchmarks/android/gradle/wrapper/gradle-wrapper.properties

```diff
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-all.zip
```

Comment:
Are you updating to these versions of Gradle and AGP (8.4, 8.1.0) in the tests because of https://docs.gradle.org/current/userguide/compatibility.html#android?

Reply:
I'm updating them because of b/342522139 - basically `compileSdk 35` seems to require AGP 8.1.0+. The specific choice of Gradle 8.4 was arbitrary though.

Reply:
I also realized that because this changes the `FlutterExtension`, it would also impact existing apps, so I added a gradle error handler for this case and also added a test for it.

Example of the error message:
```
Execution failed for task ':app:processReleaseResources'.
> A failure occurred while executing com.android.build.gradle.internal.res.LinkApplicationAndroidResourcesTask$TaskAction
   > Android resource linking failed
     aapt2 E 08-19 15:13:02 78026 5943581 LoadedArsc.cpp:94] RES_TABLE_TYPE_TYPE entry offsets overlap actual entry data.
     aapt2 E 08-19 15:13:02 78026 5943581 ApkAssets.cpp:152] Failed to load resources table in APK '/Users/mackall/Library/Android/sdk/platforms/android-35/android.jar'.
     error: failed to load include path /Users/mackall/Library/Android/sdk/platforms/android-35/android.jar.
```
And example of the handled version:
```
┌─ Flutter Fix ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ [!] Using compileSdk 35 requires Android Gradle Plugin (AGP) 8.1.0 or higher.                                                              │
│ Please upgrade to a newer AGP version. The version of AGP that your project uses is likely defined in:                                     │
│ /Users/mackall/development/BugTesting/testFlutterPluginGradlePlugin/foobarfoo/android/settings.gradle,                                     │
│ in the 'plugins' closure.                                                                                                                  │
│ Alternatively, if you have a strong reason to avoid upgrading AGP, you can temporarily lower the compileSdk version in the following file: │
│ /Users/mackall/development/BugTesting/testFlutterPluginGradlePlugin/foobarfoo/android/app/build.gradle                                     │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

Reply:
Hmm I agree it is the less likely one (hopefully not by much now?), but it is definitely the one we support the most. I can keep telling people both locations for now I suppose.

Do you have any sense of when we would stop providing both links in situations like these? A year from now?

Reply:
Example of how it looks now:
```
┌─ Flutter Fix ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ [!] Using compileSdk 35 requires Android Gradle Plugin (AGP) 8.1.0 or higher.                                                         │
│  Please upgrade to a newer AGP version. The version of AGP that your project uses is likely defined in:                               │
│ /Users/mackall/development/BugTesting/testFlutterPluginGradlePlugin/foobarfoo/android/settings.gradle,                                │
│ in the 'plugins' closure.                                                                                                             │
│  Alternatively, if your project was created with an older version of the templates, it is likely                                      │
│ in the buildscript.dependencies closure of the top-level build.gradle:                                                                │
│ /Users/mackall/development/BugTesting/testFlutterPluginGradlePlugin/foobarfoo/android/build.gradle.                                   │
│                                                                                                                                       │
│  Finally, if you have a strong reason to avoid upgrading AGP, you can temporarily lower the compileSdk version in the following file: │
│ /Users/mackall/development/BugTesting/testFlutterPluginGradlePlugin/foobarfoo/android/app/build.gradle                                │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

I added the new link, and also added spaces to the start of each separate item which hopefully makes it more readable?

Reply:
> Hmm I agree it is the less likely one (hopefully not by much now?), but it is definitely the one we support the most. I can keep telling people both locations for now I suppose, though I feel that three file paths in the message makes it a bit lengthy/gross.
> 
> Do you have any sense of when we would stop providing both links in situations like these? A year from now?

We would stop telling people about both when we have removed the ability to import FGP as a script. Maybe a year from now. Most people have not migrated, my gut say 90+ percent of people. I could do a pub example check if we wanted.